### PR TITLE
Added BX-out ports.

### PR DESCRIPTION
**Pull request by aehart**
This PR adds BX-out ports to the TE and TC, as has already been done for the PR, ME, and MC.

I would suggest that @sseifeln and @meisonlikesicecream add similar ports to the IR and VMR and push those changes to this same branch when they're ready.

### CODE REVIEW

**Comment by sseifeln** [comment]
@aehart - @meisonlikesicecream commit reminded me that I had to this :)

**Comment by aehart** [comment]
Thanks @sseifeln and @meisonlikesicecream! This looks good to me.

@djcranshaw, could you take a gander and make sure this looks good to you too?

**Review by djcranshaw — COMMENTED** [review]
For IR, VMR, TE, TC, we have all updated the bx_o after the step loop is completed. This is how it's done in ME, but not how it's done in PR and MC, which update the bx_o on the last iteration of the step loop. I've checked and there is no difference in timing, II, or resource usage, and according to the vivado_hls waveform viewer, there is no difference in the relative timing between the HLS-generate done, idle, and ready signals, and the bx_o between these two scenarios. For consistency across the project, I will add a commit to this Pull Request to bring PR and MC into agreement with the rest of the modules. After that's done, and the IR variable-name change requested above, this PR is good to go.

File: TestBenches/InputRouter_test.cpp [hunk 1]
```diff
@@ -373,6 +373,7 @@ int main()
     static const int* cLUT_L3 = (  hIs2S == 1 )  ? kPhiCorrtable_L6 : kPhiCorrtable_L3; 
 
     BXType hBx = cEvId&0x7;
+    BXType hBxOut;
```
[the remark this answers is not in the document]
Comment (sseifeln) [reply TestBenches/InputRouter_test.cpp:376]: @djcranshaw - sure. also in the test bench or only in the IR functions?

[the remark this answers is not in the document]
Comment (sseifeln) [reply TestBenches/InputRouter_test.cpp]: wasn't fast enough?

[the remark this answers is not in the document]
Comment (aehart) [reply TestBenches/InputRouter_test.cpp]: @sseifeln, looks like you missed one reference to `hOutputBx` in the pragma on line 144 of InputRouter.h.

[the remark this answers is not in the document]
Comment (sseifeln) [reply TestBenches/InputRouter_test.cpp]: Argh. Sorry . did not grep all the things. Fixing it now ..

**Review by djcranshaw — APPROVED** [review]
Thanks @sseifeln. I think this PR is ready to merge.

**Pull request merged**